gluefactory/datasets/merge_datasets.py

```diff
@@ -60,11 +66,26 @@ def __init__(self, conf, split):
         #self.set_split_for_all_datasets(split)
         self.datasets = {}  # store dataset objects
         self.img_index_collection = []  # store image indices
+        if self.conf.use_multiscale_learning:
+            self.relevant_batch_size = self.conf[f"{split}_batch_size"]
+            self.num_selected_this_batch = 0
+            self.current_scale = None
+            
 
         logging.info(f"Initialize Merged Dataset with following datasets: {conf['datasets'].keys()}")
         for key, dset_conf in conf["datasets"].items():
+            # 1st check if mulitscale learning scale selection is random for all
```

Comment:
I think this whole code block would be a lot simpler if the multi-scale configuration wasn't on the level of an individual dataset. Do you think it would be possible to implement this? If it's too complicated it's also fine by me for the purposes of the paper

Reply:
I think we need multiscale training also on single dataset level. At least for the ablation.

Or are you suggesting a change in implementation?

Reply:
Sure, I agree that we need it on a single dataset level. But maybe it is possible to override the single dataset configuration in case of having the merged dataset?

Reply:
You mean deactivating multiscale training for the datasets that are being merged and handle it in the merged_dataset class? 

This would require more logic on the merged_dataset class. In this case I tried to resuse logic as much as possible. But this could be a consideration for a refactoring when we design a comnmon multiscale training superclass :)

Will add this to the issue :)